File: src/models/movie.py
from sqlalchemy import Column, Integer, String, ForeignKey
from src.models.base import Base
from dataclasses import dataclass


@dataclass
class Movie(Base):
    __tablename__ = 'movies'

    id: str = Column(String, primary_key=True)
    name: str = Column(String)
    release_date: str = Column(String)
    genre: str = Column(String)
    status: str = Column(String)
    filename: str = Column(String)
    video_codec: str = Column(String)
    overview: str = Column(String)
    size: int = Column(Integer)
    space_saved: int = Column(Integer)
    profile_id: int = Column(Integer, ForeignKey('profiles.id'))
    monitored: bool = Column(Integer)
    missing: bool = Column(Integer)
    studio: str = Column(String)
    original_size: int = Column(Integer)
    path: str = Column(String)
    runtime: int = Column(Integer)

File: internal/models/movie.go
package models

type Movie struct {
	Id           string `json:"id"`
	Name         string `json:"name"`
	ReleaseDate  string `json:"releaseDate"`
	Genre        string `json:"genre"`
	Status       string `json:"status"`
	Filename     string `json:"filename"`
	VideoCodec   string `json:"videoCodec"`
	Overview     string `json:"overview"`
	Size         int    `json:"size"`
	SpaceSaved   int    `json:"spaceSaved"`
	ProfileID    int    `json:"profileId"`
	Monitored    bool   `json:"monitored"`
	Missing      bool   `json:"missing"`
	Studio       string `json:"studio"`
	OriginalSize int    `json:"originalSize"`
	Path         string `json:"path"`
	Runtime      int    `json:"runtime"`
	FileID       string `json:"fileId"`
	File         *File  `json:"file"`
}
